imdbfunction.py

```diff
@@ -0,0 +1,29 @@
+from bs4 import BeautifulSoup
+from bs4.dammit import EncodingDetector
+import requests
+import re
+
+
+class ImdbFunction :
+
+    def getIMDB(self, queryi) :
+ 
+
+        resp = requests.get('https://www.imdb.com/search/keyword/?keywords=' + str(queryi) + '&ref_=fn_kw_kw_1&mode=detail&page=1&sort=moviemeter,asc')
+        http_encoding = resp.encoding if 'charset' in resp.headers.get('content-type', '').lower() else None
+        html_encoding = EncodingDetector.find_declared_encoding(resp.content, is_html=True)
+        encoding = html_encoding or http_encoding
+        soup = BeautifulSoup(resp.content, from_encoding=encoding,features="lxml")
+        
+        imdbLinkList = []
+        
+        #find links to titles
+        for link in soup.find_all('a', href= re.compile('title')) :
+            #remove irrelevant links
+            if "vote" not in link['href'] and "search" not in link['href'] and "plotsummary" not in link['href'] :
+                #remove duplicates
+                if('https://www.imdb.com' + link['href'] not in imdbLinkList) :

```

Comment:
Optionally, you could just add these to a `set()` and you wouldn't need this if logic

Reply:
That's a good idea, but the links in this list have a specific order which sets won't guarantee upon creation. I am still open to and looking for better logic for this block of code.